syftlib/src/test/java/org/openmined/syft/unit/monitor/network/NetworkStatusRepositoryTest.kt

```diff
@@ -0,0 +1,51 @@
+package org.openmined.syft.unit.monitor.network
+
+import android.net.NetworkCapabilities
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.spy
+import com.nhaarman.mockitokotlin2.stub
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Flowable
+import io.reactivex.Maybe
+import io.reactivex.processors.PublishProcessor
+import org.junit.Test
+import org.openmined.syft.common.AbstractSyftWorkerTest
+import org.openmined.syft.common.HttpClientMock
+import org.openmined.syft.common.SocketClientMock
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.monitor.StateChangeMessage
+import org.openmined.syft.monitor.network.NetworkStatusCache
+import org.openmined.syft.monitor.network.NetworkStatusModel
+import org.openmined.syft.monitor.network.NetworkStatusRealTimeDataSource
+import org.openmined.syft.monitor.network.NetworkStatusRepository
+import org.openmined.syft.networking.clients.NetworkMessage
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRepositoryTest {
+
+    @Test
+    fun `NetworkStatusRepository subscribe successfully`() {
+        val networkStatusRealDataSource = mock<NetworkStatusRealTimeDataSource> {
+            on { subscribeStateChange() }.thenReturn(Flowable.just(StateChangeMessage.NetworkStatus(true)))
+        }
+
+        val networkStatusCache = mock<NetworkStatusCache>()
+        val networkStatusRepository = NetworkStatusRepository(listOf(), networkStatusCache, networkStatusRealDataSource)
+        networkStatusRepository.subscribeStateChange().test().hasSubscription()
+    }
+
+    @Test
+    fun `network validity call NetworkStatusDataSource's getNetworkValidity correctly`() {
+        val networkStatusRealDataSource = mock<NetworkStatusRealTimeDataSource> {
+            on { getNetworkValidity(listOf()) }.thenReturn(true)
+        }
+        val networkStatusRepository = NetworkStatusRepository(listOf(), mock(), networkStatusRealDataSource)
+
+        assert(networkStatusRepository.getNetworkValidity())
```

Comment:
Instead of assert here better call verify on the mocked network status datasource